examples/lock-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "CHIPDeviceManager.h"
+#include "DataModelHandler.h"
+#include "DeviceCallbacks.h"
+#include "Server.h"
+#include "esp_heap_caps_init.h"
+#include "esp_log.h"
+#include "esp_netif.h"
+#include "esp_spi_flash.h"
+#include "esp_system.h"
+#include "esp_wifi.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "nvs_flash.h"
+
+#include <cmath>
+#include <cstdio>
+#include <string>
+#include <vector>
+
+#include <support/ErrorStr.h>
+using namespace ::chip;
+using namespace ::chip::DeviceManager;
+using namespace ::chip::DeviceLayer;
+
+static const char * TAG = "lock-app";
+
+static DeviceCallbacks EchoCallbacks;
+
+extern "C" void app_main()
+{
+    int err = 0;
+    // Initialize the ESP NVS layer.
+    err = nvs_flash_init();
+    if (err != CHIP_NO_ERROR)
+    {
+        ESP_LOGE(TAG, "nvs_flash_init() failed: %s", ErrorStr(err));
+        return;
+    }
+
+    ESP_LOGI(TAG, "==================================================");
+    ESP_LOGI(TAG, "chip-esp32-lock-example starting");
+    ESP_LOGI(TAG, "==================================================");
+
+    CHIPDeviceManager & deviceMgr = CHIPDeviceManager::GetInstance();
+
+    err = deviceMgr.Init(&EchoCallbacks);
+    if (err != CHIP_NO_ERROR)
+    {
+        ESP_LOGE(TAG, "device.Init() failed: %s", ErrorStr(err));
+        return;
+    }
+
+    InitServer();
+
+    // Run the UI Loop
+    while (true)
+    {
+        vTaskDelay(50 / portTICK_PERIOD_MS);
```

Comment:
I believe this comment (`Run the UI Loop`) and 50ms loop delay have been propagated from esp32 example projects that do actually service a UI in main.

But since there's nothing serviced here, the comment is misleading.  Also, this delay could be increased a lot.  There's no reason to wake every 50ms.

Reply:
@sweetymhaiske ?

Reply:
@msandstedt @woody-apple  Done. Removed the comment and increased the delay to 500 ms.